### PR TITLE
Add local account setup flow

### PR DESCRIPTION
**Pull request by AdamLinCodes**
This is Hugo, Adam’s OpenClaw agent, submitting this PR.

## Summary
- add a first-run / migration-safe local account setup flow for new and existing users
- persist account metadata alongside local app data so future database sync and user-to-user communication work has a stable profile foundation
- keep reset-app-data behavior focused on clearing goals/history while preserving the account profile, and add unit tests for the new account helpers/store behavior

## Edge cases covered
- existing users without an account are prompted on launch and keep their current local data
- new users still get demo-data onboarding first, then set up an account before entering the app
- usernames are normalized from entered text and default sensibly from the display name
- reset app data clears goals/history but preserves the account profile for future sync/communication identity

## Validation
- `npm test -- --runInBand`
- `npx tsc --noEmit`

## Checkout
`git fetch && git checkout hugo/issue-91-setup-user-accounts`

### CODE REVIEW

File: account.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  isValidUsername,
+  normalizeAccountDraft,
+  passwordsMatch,
+  sanitizeUsername,
+  shouldPromptForAccountSetup,
+} from './account';
+
+describe('account helpers', () => {
+  it('builds a stable default username from a display name', () => {
+    expect(buildDefaultUsername(' Adam Lin ')).toBe('adam-lin');
```
Comment (AdamLinCodes) [inline account.test.ts:17]: Not sure if this is true

Comment (AdamLinCodes) [reply account.test.ts:17]: Fixed
